server/queue.php

```diff
@@ -3,20 +3,46 @@
 	$conn = $DB_CONN;
 
 	// TODO make this handle multiple locations, if necessary
-	$stmt = $conn->prepare('SELECT id, time, firstName, lastName FROM Appointment
-		WHERE (date = getdate() && archived = 0)');
+	$stmt = $conn->prepare('SELECT DISTINCT appointmentId, scheduledTime
+		FROM Appointment
+		WHERE (Appointment.scheduledTime >= NOW() AND Appointment.scheduledTime < DATE_ADD(CURDATE(), INTERVAL 1 DAY))
+			AND Appointment.archived = FALSE
+		ORDER BY Appointment.scheduledTime ASC');
+
 	$stmt->execute();
 	$results = $stmt->fetchAll();
 
 	// We must only display the first letter of the last name
 	// We do this server-side since we can't disclose the data client-side
 	$appointments = [];
 	foreach ($results as $result) {
-		$result['lastName'] = substr($result['lastName'], 0, 1);
+		$firstNameStmt = $conn->prepare('SELECT Answer.string AS firstName
+			FROM AppointmentQuestionAnswer AS AQA
+			JOIN Answer ON AQA.answerId = Answer.answerId
+			JOIN Question ON Answer.questionId = Question.questionId
+			WHERE Question.tag = "first_name" AND AQA.appointmentId = ?');
+
+		$lastNameStmt = $conn->prepare('SELECT Answer.string AS lastName
+			FROM AppointmentQuestionAnswer AS AQA
+			JOIN Answer on AQA.answerId = Answer.answerId
+			JOIN Question ON Answer.questionId = Question.questionId
+			WHERE Question.tag = "last_name" AND AQA.appointmentId = ?');
```

Comment:
With setting them null after the foreach block as well. Is it safer in some way to prepare the statement multiple times? I thought a prepared statement was to avoid that?